core/templates/dev/head/pages/topic-editor-page/questions-tab/topic-questions-tab.directive.ts

```diff
@@ -89,6 +91,11 @@ angular.module('oppia').directive('questionsTab', [
             $scope.emptyMisconceptionsList = [];
           };
 
+          $scope.reinitializeQuestionsList = function(skillId) {
+            $scope.$broadcast('changeSelectedSkill', skillId);
```

Comment:
As noted earlier, try not to use a broadcast/on structure if it can be avoided.

Reply:
Done